packages/router/src/lib/entities/cache.ts

```diff
@@ -0,0 +1,23 @@
+import { BigNumber } from "ethers";
+
+export type StoreOutstandingLiquidityParams = {
+  assetId: string;
+  chainId: number;
+  amount: BigNumber;
+  expiresInSeconds: number;
+  transactionId: string;
+};
+
+export type GetOutstandingLiquidityParams = { assetId: string; chainId: number };
+
+export type RemoveOutstandingLiquidityParams = {
+  transactionId: string;
+  assetId: string;
+  chainId: number;
+};
+
+export type Cache = {
+  storeOutstandingLiquidity: (s: StoreOutstandingLiquidityParams) => Promise<void>;
+  getOutstandingLiquidity: (g: GetOutstandingLiquidityParams) => Promise<BigNumber>;
+  removeOutstandingLiquidity: (r: RemoveOutstandingLiquidityParams) => Promise<void>;
+};
```

Comment:
How are you using this type to check if you have the liquidity to respond to an auction? Seems like we'd have to pull all of them, calculate funds, then we'd know how much we have left to offer. Don't love the "pull all of them" step

Reply:
The function hides the implementation of how you calculate the outstanding liquidity. I'm not sure I understand exactly what your concern is, but you can see from the implementation of the `getOutstandingLiquidity` function that it does pull all the saved bids and add them up.